vars/piperPipelineStageInit.groovy

```diff
@@ -106,7 +115,10 @@ void call(Map parameters = [:]) {
     def stageName = StageNameProvider.instance.getStageName(script, parameters, this)
 
     piperStageWrapper (script: script, stageName: stageName, stashContent: [], ordinal: 1, telemetryDisabled: true) {
-        def scmInfo = checkout(parameters.checkoutMap ?: scm)
+        def scmInfo = parameters.scmInfo
+        if (!parameters.skipCheckout) {
+            scmInfo = checkout(parameters.checkoutMap ?: scm)
+        }
```

Comment:
In case there is no checkout and `parameters.scmInfo` is not provided `scmInfo` will be `null`. That is most likely a problem.

Reply:
I checked, it is no problem for setupCPE to get a null as scmInfo. [Here it is checked in setupCPE](https://github.com/SAP/jenkins-library/blob/master/vars/setupCommonPipelineEnvironment.groovy#L118)

Reply:
How about setting default value to this parameter in `default_pipeline_environment.yml`?

Reply:
Ok, I see. But nevertheless up to now `scmInfo` was never `null`. So we might have some problem inside GPP caused by some git settings which are silently disappearing now. Not sure if the GPP expects these values. Maybe working without these git settings is more something outside the GPP and maybe those other use cases are the reason for the null check for `scmInfo` in `setupCommonPipelineEnvironment` I feel not very comfortable ...

Reply:
@srinikitha09 : but what happens when someone in the project config or at any other layer closer to the pipeline overwrites it with a string?

Reply:
May be in that case it should be mentioned in the parameter description that it of boolean type.

Reply:
@srinikitha09 ... nevertheless it is a good idea to have that parameter in our default layer. We should add it with boolean value `false`. I mean it behaves in the same way without, but that would be more explicit. Forgot to mention that in my earlier comment.

Reply:
> May be in that case it should be mentioned in the parameter description that it is of boolean type.

I will add the `@possibleValues` annotation with the values `true` and `false`

Reply:
> How about setting default value to this parameter in `default_pipeline_environment.yml`? for ex: `skipCheckout: false`

Will do so :+1

Reply:
> I will add the @possibleValues annotation [...]

That is better anyway. However it will still happen that someone provides a string `"false"` in the config. That's hard to trouble shoot. Having the possible values annotation in the docu helps us to blame that bad guy, but in fact the troubleshooting effort remains. Not against that as solution, but having a real boolean would be better IMO.

Reply:
We should not forget about the `scmInfo` which can now be null - which was not like that before. I still doubt that having that with value `null` is something what should happen inside GPP.

Reply:
The existence of a scmInfo map is now checked when wanting to skip the checkout.